### DISCUSSION TITLE
Any chance to add Temperature sensors?

### DISCUSSION THREAD

**Discussion by 000al000** [odya/esphome-powmr-hybrid-inverter#12] [Q&A]
In other project there are two additional sensors

 "Inv Temperature": {'register':545, 'multiplier':0.1, 'unit':'C'},
"Ambient Temperature": {'register':546, 'multiplier':0.1, 'unit':'C'},

Any chance to add this sensors in this project?

**Comment by odya** [comment]
I'm not receiving any data from these registers on my `POW-HVM2.4H-24V`. I'm aware that there is a pair of thermal sensors in my inverter, but I believe they are "dumb," so you can't actually retrieve any data from them.

Maybe in your model situation is different😉

**Comment by 000al000** [comment]
I have POW-HVM6.2M-48V.
I dont understand how i can translate register from 'register':545,, to you registers :)
For example i try to compare "Battery voltage" register, but they are completle different

**Reply by heartpolik** [reply]
Do you solve this problem?

**Comment by odya** [comment]
I think that register numbers must be the sae. You can try to extract data from register 545 by adding test sensor:
```
  - platform: modbus_controller
    modbus_controller_id: smg_inverter
    name: "Temp Test"
    address: 545
    register_type: holding
    value_type: U_WORD
    lambda: |-
      return swapBytes(x);
```
